web/source/osk/visualKeyboard.ts

```diff
@@ -701,77 +697,29 @@ namespace com.keyman.osk {
         return;
       }
 
-      // Do not move over keys if device popup visible
-      if(this.popupVisible) {
-        if(key1 == null) {
-          if(key0) {
-            this.highlightKey(key0,false);
-          }
-          this.keyPending=null;
-          this.touchPending=null;
-        } else {
-          if(key1 == this.popupBaseKey) {
-            if(!key1.classList.contains('kmw-key-touched')) {
-              this.highlightKey(key1,true);
-            }
-            this.keyPending = key1;
-            this.touchPending = e.touches[0];
-          } else {
-            if(key0) {
-              this.highlightKey(key0,false);
-            }
-            this.keyPending = null;
-            this.touchPending = null;
-          }
-        }
```

Comment:
Moved into the `SubkeyDelegator.updateTouch` method, as this block was solely responsible for embedded-mode subkey highlighting updates. Also note that the `if` block is a perfect match with the inner `else` block, so the only real conditional is the `else -> if`.